sphinx_gallery/tests/test_full.py

```diff
@@ -62,3 +62,9 @@ def test_embed_links(sphinx_app):
     assert 'numpy.arange.html' in lines
     # assert '#module-matplotlib.pyplot' in lines
     # assert 'pyplot.html' in lines
+
+
+def test_clean(sphinx_app):
```

Comment:
Do we want to make sure this function is run as the last test, otherwise you may break other tests by cleaning some of the generated artifacts?

I don't know off the top of my head if there is a way to ensure that with pytest but it's worth looking at it. Maybe a module teardown could work also that seems like a hack.

Reply:
Can't find it the docs right now but I am almost certain pytest runs the tests in the source order (the fact that the tests pass on travis despite e.g. hash randomization on Py3 clearly indicates (well, strongly suggests :-)) that that's the case).

Reply:
OK the only thing I found is [this SO answer](https://stackoverflow.com/a/44867455). I tested this a bit and the only tricky case where ordering does not seem to be respected is if you use a class test that derives from unittest.TestCase.

If that ends up being too brittle, we can always try to use [pytest-ordering](https://github.com/ftobia/pytest-ordering).